### PR TITLE
ci: move assert-test-linux to GitHub Actions

### PR DESCRIPTION
**Pull request by aykevl**
This is needed since CircleCI has imposed a limit of 60 minutes on the length of a job - even if that happens only infrequently.

### CODE REVIEW

**Comment by niaow** [comment]
This looks good so far

**Comment by aykevl** [comment]
I'm having some trouble with the wasm tests. They're timing out sometimes.

**Comment by aykevl** [comment]
So I actually forgot to add the `ASSERT=1`, which is the whole point of the assert-test-linux job...

This PR should now be more or less ready. I have a few CI jobs commented out during development because I don't need them, I'll update them once this PR is ready.

My idea is to move at least the Linux jobs over GitHub Actions, but not all at once to make it easier to review and to minimize disruption in the process. (Although `build-linux` and `test-linux-build` will need to be moved at once).

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  assert-test-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Install apt dependencies
+        run: |
+          sudo apt-get install --no-install-recommends \
```
Comment (niaow) [inline .github/workflows/linux.yml:20]: Is apt-get update no longer necessary here?

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: Windows
 
 on:
-  pull_request:
```
Comment (niaow) [inline .github/workflows/windows.yml]: Was this just temporary for debugging?

Comment (aykevl) [reply .github/workflows/windows.yml]: Yes. I'll remove it before marking this PR as ready.

File: .github/workflows/linux.yml (hunk 1 above)
Comment (aykevl) [reply .github/workflows/linux.yml:20]: Apparently not.
And if it is, we can always add it back.

**Review by niaow — APPROVED** [review]
This mostly looks good to me. I can't seem to find anything wrong with it now. It does seem a bit slower in a few places, but not by that much.

**Comment by deadprogram** [comment]
Merging. Thanks @aykevl and @niaow

**Pull request merged**